core/controllers/classifier.py

```diff
@@ -79,7 +79,10 @@ def verify_signature(message, vm_id, received_signature):
     if secret is None:
         return False
 
-    generated_signature = generate_signature(secret, message)
+    if message is None:
```

Comment:
So all the recent discussion boils back down to this comment:

> There is still one problem, though. I didn't notice it earlier but if we go with message then signature will be generated from message part but since that is empty, signature is also going to be empty and we don't have the security now. We need something in message.

I just tested this, and now I'm not sure if I understand the issue. A hash is still generated even if there's nothing in the dict:

```
import json
import hmac
import hashlib

message_json = json.dumps({})
print message_json
>>> {}

print hmac.new('asd', message_json, digestmod=hashlib.sha256).hexdigest()
>>> dd9f5b5ddfc902dbde59ddd348c94d3b2a5dd0b39ca6bacf006fdbb5019b63f2
```